### PR TITLE
Formatting of ModelChain.diode_params in pandas.DataFrame

### PR DESCRIPTION
**Pull request by tylunel**
- [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

This PR is for changing the type of the attribute ModelChain object `diode_params` from tuple to pandas.DataFrame. It would standardize the attributes of ModelChain, since all array-like attributes are either pandas.DataFrame or pandas.Series, except diode_params which is a tuple. Moreover it simplifies the use of the diode_params attribute for using it in other simulations (like simuling the direct coupling with an electrical load).

### CODE REVIEW

**Comment by wholmgren** [comment]
No objection from me, but this is a breaking API change. Not sure if it's worth deprecating or not.

**Comment by cwhanse** [comment]
Thanks @tylunel this is a good idea, it links the values for the diode parameters to the inputs via the index. I'm OK proceeding without deprecation. My guess is that those using `ModelChain.diode_params` (instead of the `PVSystem` methods and functions) will appreciate having the index to make plotting and slicing easier.

**Comment by tylunel** [comment]
Thanks for your feedbacks.
I feel like there is no relevant test to add on it, and the api.rst is too global for noticing a type change in it. Therefore I checked the check boxs 'test added' and 'Update entries to api.rst' whereas I didn't do it particularly (didn't found a way to remove irrelevant check boxs after PR was pushed). 
If you agree on that, my PR is complete. If not I am open to any suggestion!

**Comment by cwhanse** [comment]
@wholmgren let's include this in v0.7

File: pvlib/modelchain.py [hunk 1]
```diff
@@ -926,7 +929,7 @@ def run_model(self, weather, times=None):
 
         Assigns attributes: solar_position, airmass, irradiance,
         total_irrad, effective_irradiance, weather, cell_temperature, aoi,
-        aoi_modifier, spectral_modifier, dc, ac, losses.
+        aoi_modifier, spectral_modifier, dc, ac, losses, diode_params.
```
Comment (wholmgren) [inline pvlib/modelchain.py]: This attribute is only assigned for single diode methods. Not sure if/how to document that.

File: docs/sphinx/source/whatsnew/v0.7.0.rst [hunk 2]
```diff
@@ -108,6 +108,8 @@ API Changes with Deprecations
   argument for capability during the deprecation period.
 * Parameter `pvsystem.DC_MODEL_PARAMS` is renamed to `pvsystem._DC_MODEL_PARAMS`.
   Users should not rely on this dictionary's existence or structure.
+* The `modelchain.ModelChain.diode_params` attribute is now formatted in
+  a pandas.DataFrame with `DatetimeIndex`, rather than in a tuple.
```
Comment (wholmgren) [inline docs/sphinx/source/whatsnew/v0.7.0.rst]: This is currently in the `API Changes with Deprecations` section of the whats new file. It should be moved to the `API Breaking Changes` section.

File: pvlib/modelchain.py (hunk 1 above)
Comment (cwhanse) [inline pvlib/modelchain.py]: perhaps ", diode_params (if dc_model is a single diode model)" 

or in a new list:

Optionally assigns attributions: diode_params if dc_model is a single diode model.

**Comment by cwhanse** [comment]
@tylunel thanks again!

**Pull request merged**